File: R/p1_hello_llm.R
# p1_hello_llm.R

library(dotenv)
dotenv::load_dot_env(file = ".env")  # no override argument in your dotenv

api_key <- Sys.getenv("OPENAI_API_KEY")
if (api_key == "") stop("OPENAI_API_KEY is not set. Put it in .env (or set an env var).")

prompt <- "Hello, how are you on this fine day?"
body <- list(
  model = "gpt-4o-mini",
  messages = list(
    list(role = "user", content = prompt)
  )
)

library(httr2)
library(jsonlite)

req <- request("https://api.openai.com/v1/chat/completions") |>
  req_headers(
    Authorization = paste("Bearer", api_key),
    `Content-Type` = "application/json"
  ) |>
  req_body_json(body)

resp <- req_perform(req)

status <- resp_status(resp)
text <- resp_body_string(resp)

if (status < 200 || status >= 300) {
  stop(sprintf("OpenAI API request failed (HTTP %s):\n%s", status, text))
}

json <- fromJSON(text, simplifyVector = FALSE)

assistant_text <- json$choices[[1]]$message$content
cat(assistant_text, "\n")

File: p1_hello_llm.py
#Loads env variables from the .env file into Python's os environment
from dotenv import load_dotenv

#Import the OpenAI client class - this is what makes API calls
from openai import OpenAI

# load the .env file into Python's os environment
load_dotenv(override=True)

#create the client - it autommactially reads OPENAI_API_KEY from the environment
client = OpenAI()

#Make the API call -- this is the actual HTTP POST to OpenAI's servers
response = client.chat.completions.create(
    model="gpt-4o-mini", # The model to use for the API call
    messages=[ #The messae list - this is the conversation you're sending 

    #one message to the user with a hardcoded prompt
    {"role": "user", "content": "Hello, how are you on this fine day?"}
    ]
)

# Navigate the response object to extrat just the tet reply
print(response.choices[0].message.content)